### PR TITLE
Improve Experience Points History view

### PR DESCRIPTION
**Pull request by kxmbrian**
- Add the last updated time of experience points records
- Since we display the "updated at" column, show the updater instead of creator so that it is consistent
- Use assessment titles as the experience points award reason for submissions

Future PRs:
- Allow users to click on the reason and view the points awarding item

### CODE REVIEW

File: lib/extensions/acts_as_helpers/active_record/base.rb [hunk 1]
```diff
@@ -18,5 +18,14 @@ module ExperiencePointsInstanceMethods
...
       false
     end
+
+    # The description to be displayed in the 'reason' column for the experience points record on
+    # the experience points history page.
+    # This method is meant to be overridden by each class that acts as an experience points record.
+    #
+    # @return [String] Reason which will be displayed
+    def points_display_reason
+      raise NotImplementedError
```
Comment (lowjoel) [inline lib/extensions/acts_as_helpers/active_record/base.rb]: Give a reason why this is raised so people can grep the source code to figure this call site.

File: app/helpers/course/experience_points_records_helper.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+# frozen_string_literal: true
+module Course::ExperiencePointsRecordsHelper
+  # Formats the reason to display for the given experience points record.
+  #
+  # @param [Course::ExperiencePointsRecord] record The experience points record to be displayed
+  # @return [String] The formatted reason for the award
+  def display_reason(record)
```
Comment (lowjoel) [inline app/helpers/course/experience_points_records_helper.rb]: this should be a model method

File: app/models/concerns/course/assessment/submission/experience_points_display_concern.rb [hunk 3]
```diff
@@ -0,0 +1,11 @@
+# frozen_string_literal: true
+module Course::Assessment::Submission::ExperiencePointsDisplayConcern
+  extend ActiveSupport::Concern
+
+  # The reason to be displayed for the submission's experience point award.
+  #
+  # @return [String] The reason which will be displayed
+  def points_display_reason
```
Comment (lowjoel) [inline app/models/concerns/course/assessment/submission/experience_points_display_concern.rb]: for consistency with the rest of the API, `experience_points_record_reason`

File: app/models/course/experience_points_record.rb [hunk 4]
```diff
@@ -23,4 +23,11 @@ def active?
   def manually_awarded?
     actable_type.nil? && actable.nil?
   end
+
+  # The string to be displayed under the 'reason' column in the experience points history page.
+  #
+  # @return [String] The formatted reason for the award
+  def display_reason
```
Comment (lowjoel) [inline app/models/course/experience_points_record.rb]: Actually, why not just override `#reason`?

Comment (kxmbrian) [inline app/models/course/experience_points_record.rb]: I went ahead and overrode it in the latest push

File: spec/libraries/acts_as_exp_record_spec.rb [hunk 5]
```diff
@@ -16,4 +16,5 @@ def self.columns
   it { is_expected.to respond_to(:course_user) }
   it { is_expected.to respond_to(:acting_as) }
   it { expect(subject.acting_as).to respond_to(:specific) }
```
Comment (lowjoel) [inline spec/libraries/acts_as_exp_record_spec.rb]: this should use the it with a do-end block.

File: spec/libraries/acts_as_exp_record_spec.rb [hunk 6]
```diff
@@ -16,4 +16,5 @@ def self.columns
...
+  it { expect { subject.experience_points_display_reason }.to raise_error(NotImplementedError) }
```
Comment (lowjoel) [inline spec/libraries/acts_as_exp_record_spec.rb]: this too.

**Comment by lowjoel** [comment]
@kxmbrian Fix the comments in a separate PR.

**Pull request merged**